xbmc/guilib/GUIWindowManager.cpp

```diff
@@ -779,6 +781,28 @@ void CGUIWindowManager::ActivateWindow_Internal(int iWindowID, const std::vector
     return;
   }
 
+  if (iWindowID == WINDOW_VIDEO_NAV)
+  {
+    if (params.front().substr(0, 4) != "root" && params.front().substr(0, 10) != "library://")
```

Comment:
IIRC we have `StringUtils::StartsWith()` to avoid possible bugs in manual string length calculation

Reply:
done